bindings/rust/s2n-tls-hyper/src/connector.rs

```diff
@@ -118,7 +126,11 @@ where
             return Box::pin(async move { Err(Error::InvalidScheme) });
         }
 
-        let builder = self.conn_builder.clone();
+        // Attempt to negotiate HTTP/2.
+        let builder = connection::ModifiedBuilder::new(self.conn_builder.clone(), |conn| {
+            conn.set_application_protocol_preference([b"h2"])
+        });
```

Comment:
So if I'm understanding this right, we'll always attempt to negotiate http2 and there's no way to choose to negotiate http1. That sounds reasonable to me given the age of http1, but I'm not an http expert-- is it reasonable? How does vanilla hyper behave?

Reply:
Correct - currently it will always attempt to negotiate http2 without a way to choose http1. However, if someone needs a way to disable http2 we can always provide an option for that in the future.

It sounds reasonable to me as well, but I'm also not an http expert. Do you have an opinion, @jmayclin?

HTTP/2 was designed to be negotiated with TLS via the ALPN. Since the vanilla HttpConnector doesn't use TLS, it doesn't negotiate HTTP/2. A better test might be looking at a TLS connector like hyper-rustls, which doesn't always negotiate HTTP/2, and instead [provides an option to enable it](https://github.com/rustls/hyper-rustls/blob/03fe2dbbc8b08b4e6bdf654c1d969cfa85134a34/src/connector/builder.rs#L241-L251). However, applications like smithy-rs just [enable this option](https://github.com/smithy-lang/smithy-rs/blob/aac9becfd469e2479f68c61fd4c8074ddf755482/rust-runtime/aws-smithy-runtime/src/client/http/hyper_014.rs#L72).